tests/test_cli.py

```diff
@@ -100,29 +117,57 @@ def verify_docker_output(docker_cmd, expect_values):
     return True
 
 
-def wait_verify_docker_output(docker_cmd, expect_values):
+def wait_verify_docker_output(docker_cmd, expect_values, use_shell=False):
     times = 0
-    while (not verify_docker_output(docker_cmd, expect_values)):
+    while (not verify_docker_output(docker_cmd, expect_values, use_shell)):
         print("Waiting until docker command is ready... ...\n")
         time.sleep(10)
         times += 1
         if times > 360:
             raise Exception('Timeout to wait until it appears expected value ' + str(expect_values))
 
 
+def get_all_docker_volumes():
+    output = start_process(['docker', 'volume', 'ls'], False)
+    return output
+
+
 def remove_docker_volumes(volumes):
     for volume in volumes:
-        start_process(['docker', 'volume', 'rm', volume, '-f'], False)
+        if volume in get_all_docker_volumes():
+            start_process(['docker', 'volume', 'rm', volume, '-f'], False)
+
+
+def get_all_docker_images():
+    output = start_process(['docker', 'image', 'ls'], False)
+    return output
 
 
 def remove_docker_images(images):
     for image in images:
-        start_process(['docker', 'rmi', '-f', image], False)
+        image_without_tag = image.split(':')[0]
+        if image_without_tag in get_all_docker_images():
```

Comment:
same, get_all_docker_images() executes for each image.

Reply:
Updated code